returnn/tf/layers/basic.py

```diff
@@ -7338,8 +7335,11 @@ def get_out_data_from_opts(cls, factor, axis, sources, name, out_dim=None, **kwa
     :rtype: Data
     """
     out = get_concat_sources_data_template(sources).copy_as_batch_major()
+    axis_ = out.get_axis_from_description(axis)
+    out = out.copy_move_axis(old_axis=axis_, new_axis=1)
     out = out.copy_template(name="%s_output" % name)
     axis = out.get_axis_from_description(axis)
+    assert axis == 1
```

Comment:
The second call to `get_axis_from_description` is not needed, and you can just assign `axis = 1` here.